File: Homework/Quang_Minh/Lesson3/RemoveAllAdjacentDuplicates.java
package HomeWorkLesson3;

import java.util.Stack;

public class RemoveAllAdjacentDuplicates {
    public static String removeDuplicates(String s) {
        Stack stack = new Stack<Character>();
        int length = s.length();
        for (int i = 0; i < length; i++) {
            if (!stack.isEmpty() && (char) stack.peek() == s.charAt(i)) {
                stack.pop();
                continue;
            }
            stack.add(s.charAt(i));
        }
        String result = "";
        while (!stack.isEmpty()) {
            result = stack.peek() + result;
            stack.pop();
        }
        return result;
    }

    public static void main(String[] args) {
        String a = "abbbbcdef";
        System.out.println(removeDuplicates(a));
    }
}

File: Homework/Van_Le/lesson3/RemoveAllAdjacentDuplicates/RemoveAllAdjacentDuplicates.cs
using System;
using System.Collections.Generic;

namespace RemoveAllAdjacentDuplicates
{
    public class RemoveAllAdjacentDuplicates
    {
        public string RemoveDuplicates(string s)
        {
            if (s.Length == 0)
                return "";

            Stack<char> charStack = new Stack<Char>();

            foreach (char ch in s)
            {
                if (charStack.Count == 0)
                {
                    charStack.Push(ch);
                    continue;
                }

                if (charStack.Peek() == ch)
                    charStack.Pop();
                else
                    charStack.Push(ch);
            }

            char[] charArray = new char[charStack.Count];

            while (charStack.Count > 0)
            {
                charArray[charStack.Count - 1] = charStack.Pop();
            }

            return new string(charArray);
        }
    }
}
